src/collection.cpp

```diff
@@ -2156,7 +2158,7 @@ Option<nlohmann::json> Collection::search(std::string raw_query,
         return Option<nlohmann::json>(422, message);
     }
 
-    size_t max_hits = DEFAULT_TOPSTER_SIZE;
+    size_t max_hits = Config::get_instance().get_default_topster_size();
```

Comment:
There is no need to parameterize `DEFAULT_TOPSTER_SIZE` because we assign max_hits in the next line this way:

```
max_hits = std::min(std::max(fetch_size, max_hits), get_num_documents());
```

Since `fetch_size` is already calculated based on the pagination depth, we will automatically have a larger max_hits value. 

So this PR only needs to parameterize `max_per_page`.